lib/jxl/field_encodings.h

```diff
@@ -19,11 +19,24 @@
 
 namespace jxl {
 
+// Macro to define the Fields' derived class Name when compiling with debug
+// names.
+#if JXL_IS_DEBUG_BUILD
+#define JXL_FIELDS_NAME(X) \
+  const char* Name() const override { return #X; }
+#else
+#define JXL_FIELDS_NAME(X)
+#endif  // JXL_IS_DEBUG_BUILD
+
 class Visitor;
 class Fields {
  public:
   virtual ~Fields() = default;
+#if JXL_IS_DEBUG_BUILD
   virtual const char* Name() const = 0;
+#else
+  const char* Name() const { return ""; };
```

Comment:
Would it be possible to just not have this method at all in non-debug builds?
(and in that case also not compile any of the code that calls it, of course)

Reply:
It is called from modular_image.cc:22 but the call is likely optimized away in non-debug builds because of an `if (false)`, however it still needs to compile. The else case is a non-virtual so the compiler doesn't need to devirtualize the method and it will likely be optimized away by the linker.

Reply:
Feel free to remove that call from modular_image, it is not useful anyway.

Reply:
(because all transforms have the name Transform)